traits/tests/test_sync_traits.py

```diff
@@ -128,6 +128,33 @@ def test_sync_lists(self):
         with self.assertTraitDoesNotChange(a, "l"):
             b.l = [7, 8]
 
+    def test_sync_lists_partial_slice(self):
```

Comment:
I was hoping to see a CI failure for this commit. I checked out this commit specifically and I can see the failure locally. FTR, the traceback is 

```

======================================================================
ERROR: Test synchronization of list traits when there is a partial slice.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/tests/test_sync_traits.py", line 150, in test_sync_lists_partial_slice
    a.l[:] = [7]
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_handlers.py", line 2568, in __setitem__
    self.name_items, TraitListEvent(index, removed, values)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_handlers.py", line 2485, in _send_trait_items_event
    object.trait_items_event(name, event, items_event)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 545, in __call__
    self.notify_listener(self, object, trait_name, old, new)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 619, in _notify_method_listener
    object, trait_name, old, new, listener
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 599, in _dispatch_change_event
    handle_exception(object, trait_name, old, new)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 180, in _handle_exception
    raise excp
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 591, in _dispatch_change_event
    self.dispatch(handler, *args)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/trait_notifiers.py", line 553, in dispatch
    handler(*args)
  File "/Users/rporuri/.edm/envs/traits-test-3.6/lib/python3.6/site-packages/traits/has_traits.py", line 2953, in _sync_trait_items_modified
    n1 = n0 + len(event.removed)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'

----------------------------------------------------------------------
```

Reply:
@rahulporuri I don't think we have any rule about using TDD for Traits development. Or am I misunderstanding your point? Is there something actionable here?

Reply:
nothing actionable. personal preference.